### PR TITLE
Pest Control

### PR DESCRIPTION
**Pull request by jamosaur**
### Description:

-   _Write description here_

### Changes:

-   _List of changes here_

-   [] I have tested all my changes thoroughly.

### CODE REVIEW

File: src/commands/Minion/pc.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		if (!selectedLanders.length) {
+			throw 'WTF?';
+		}
+
+		const { timeToFinish, points, name: landerName } = selectedLanders.reverse()[0];
+		const maxPossibleGames: number = Math.floor(
+			msg.author.maxTripLength / timeToFinish
+		) as number;
+
+		if (quantity === null || quantity === undefined) {
```
Comment (Alexsuperfly) [inline src/commands/Minion/pc.ts:53]:
```suggestion
		if (!quantity) {
```

File: src/lib/minions/data/killableMonsters/bosses/misc.ts [hunk 2]
```diff
@@ -48,7 +48,11 @@ const killableBosses: KillableMonster[] = [
 		notifyDrops: resolveItems(['Vorki', 'Jar of decay', 'Draconic visage', 'Skeletal visage']),
 		qpRequired: 205,
 		itemInBankBoosts: {
-			[itemID('Dragon warhammer')]: 10
+			[itemID('Dragon warhammer')]: 10,
+			[itemID('Elite void top')]: 10,
```
Comment (Alexsuperfly) [inline src/lib/minions/data/killableMonsters/bosses/misc.ts:52]: these seem way too high, can you do one boost for having only the whole set

File: src/commands/Minion/pc.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+			return msg.channel.send(`You have ${points} commendation points`);
+		}
+
+		const { maxTripLength, minionName } = msg.author;
+
+		const pcGames = msg.author.getMinigameScore(MinigameIDsEnum.PestControl);
+		const selectedLanders = landers.filter(({ gamesRequired }) => pcGames >= gamesRequired);
+
+		if (!selectedLanders.length) {
+			throw 'WTF?';
```
Comment (Alexsuperfly) [inline src/commands/Minion/pc.ts:45]: not sure what this is doing, either give a meaningful message if this check is valid or dont have it at all

File: src/lib/settings/types/UserSettings.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ export function T<TCustom>(k: string): CustomGet<string, TCustom> {
 export namespace UserSettings {
 	export const GP = T<number>('GP');
 	export const QP = T<number>('QP');
+	export const CommendationPoints = T<number>('commendationPoints');
```
Comment (Alexsuperfly) [inline src/lib/settings/types/UserSettings.ts:16]: reset this when the user becomes an ironman

File: src/commands/Testing/empty.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+		if (
+			this.client.production ||
+			!this.client.user ||
+			this.client.user.id === '303730326692429825'
+		) {
+			return;
+		}
+
+		await msg.author.removeGP(msg.author.settings.get(UserSettings.GP));
+		for (const [itemID, qty] of Object.entries(msg.author.settings.get(UserSettings.Bank))) {
```
Comment (Alexsuperfly) [inline src/commands/Testing/empty.ts:25]: just reset bank

File: src/commands/Minion/pc.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import { rand } from '../../util';
+import { PestControlActivityTaskOptions } from '../../lib/types/minions';
+import { MinigameIDsEnum } from '../../lib/minions/data/minigames';
+import { Activity, Tasks } from '../../lib/constants';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+
+const landers = [
+	{ name: 'novice', points: 3, gamesRequired: 0, timeToFinish: Time.Minute * 4.5 },
+	{ name: 'intermediate', points: 4, gamesRequired: 100, timeToFinish: Time.Minute * 3.1 },
+	{ name: 'veteran', points: 5, gamesRequired: 200, timeToFinish: Time.Minute * 2 }
```
Comment (Alexsuperfly) [inline src/commands/Minion/pc.ts:16]: this seems short

File: src/commands/Minion/pc.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+			cooldown: 3,
+			usage: '[quantity:int{1}]',
+			aliases: ['pestcontrol', 'pest-control'],
+			oneAtTime: true
+		});
+	}
+
+	@requiresMinion
+	@minionNotBusy
+	async run(msg: KlasaMessage, [quantity]: [null | number | string]) {
```
Comment (Alexsuperfly) [inline src/commands/Minion/pc.ts:31]: how can quantity be a string

File: src/tasks/minions/minigames/pestControlActivity.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+		const points = user.settings.get(CommendationPoints);
+		const newPoints =
+			pointsToAdd + points < MAX_PEST_CONTROL_POINTS
+				? pointsToAdd + points
+				: MAX_PEST_CONTROL_POINTS;
+		messages.push(`You now have ${newPoints} points!`);
+		if (newPoints === MAX_PEST_CONTROL_POINTS) {
+			messages.push('You have hit the maximum amount of commendation points');
+		}
+		await user.settings.update(CommendationPoints, newPoints);
```
Comment (Alexsuperfly) [inline src/tasks/minions/minigames/pestControlActivity.ts:49]: whats the point of making an addCommendationPoints method and not using it

File: src/extendables/UserExtendables.ts [hunk 9]
```diff
@@ -86,6 +86,35 @@ export default class extends Extendable {
 		return this.settings.update(UserSettings.GP, currentGP + amount);
 	}
 
+	public async addCommendationPoints(this: User, amount: number) {
```
Comment (Alexsuperfly) [inline src/extendables/UserExtendables.ts:89]: since you defined a MAX point constant, use it here

File: src/commands/Minion/pc.ts (hunk 6 above)
Comment (jamosaur) [reply src/commands/Minion/pc.ts:16]: Which part seems short here?

Average pest control game lasts around 2 minutes on the veteran boat

Comment (Lajnux) [reply src/commands/Minion/pc.ts:16]: Would store the combat lvl requirments on the landers for future and then make a temp combat variable to 126 which can be replaced in future with the users combat lvl

Comment (Alexsuperfly) [reply src/commands/Minion/pc.ts:16]: 2 mins is just about a perfect game, minions arnt perfect, and considering that this minigame would only really be played once i think increasing the time a bit to an avg of 2.5 min seems reasonable

**Pull request closed without merging**